src/Iterator/ArrayCacheIterator.php

```diff
@@ -41,6 +42,7 @@ public function __construct(Iterator $iterator)
     /**
      * @return T
      */
+    #[ReturnTypeWillChange]
     public function current()
```

Comment:
PHP 8.1 now requires the `mixed` return type: https://www.php.net/manual/en/class.iterator.php.

However, if we want Collection to continue supporting PHP 7.4 we cannot add that as it was only introduced in PHP 8. For now the workaround to silence the deprecation is to add this `#[ReturnTypeWillChange]` attribute. More details here: https://wiki.php.net/rfc/internal_method_return_types

Reply:
We could work on version 7, which would be only for php >= 8.1 !

Reply:
Hmm we could but it likely means we would need to continue supporting two versions: one for PHP 7.4 and one for >= PHP 8.1. This is because many projects will not be able to update to >= 8.0 and will continue to be on 7.4 for quite some time.

I think for now this is a good solution until PHP 7.4 reaches end of life in November 2022: https://www.php.net/supported-versions.php. We can stop supporting it in this library at that time